saleor/graphql/discount/mutations/sale/sale_channel_listing_update.py

```diff
@@ -73,18 +82,75 @@ class Meta:
         error_type_field = "discount_errors"
 
     @classmethod
-    def add_channels(cls, sale: "SaleModel", add_channels: List[Dict]):
+    def add_channels(cls, promotion: Promotion, add_channels: List[Dict]):
+        rules_data_to_add: List[RuleInfo] = []
+        rules_to_update: List[PromotionRule] = []
+
+        rules = promotion.rules.all()
+        PromotionRuleChannel = PromotionRule.channels.through
+        rule_ids = [rule.id for rule in rules]
+        rule_channel = PromotionRuleChannel.objects.filter(
+            promotionrule_id__in=rule_ids
+        )
```

Comment:
```suggestion
        
        rules = promotion.rules.all()
        PromotionRuleChannel = PromotionRule.channels.through
        rule_channel = PromotionRuleChannel.objects.filter(
            Exists(rules.filter(id=OuterRef("promotionrule_id")))
        )
```

Reply:
Does your suggestion perform better?
mine:
`SELECT "discount_promotionrule_channels"."id", "discount_promotionrule_channels"."promotionrule_id", "discount_promotionrule_channels"."channel_id" FROM "discount_promotionrule_channels" WHERE "discount_promotionrule_channels"."promotionrule_id" IN (ed26bacf-ca4c-44ae-96d9-39eacc7cb381)`

yours:
`SELECT "discount_promotionrule_channels"."id", "discount_promotionrule_channels"."promotionrule_id", "discount_promotionrule_channels"."channel_id" FROM "discount_promotionrule_channels" WHERE EXISTS(SELECT (1) AS "a" FROM "discount_promotionrule" U0 WHERE (U0."promotion_id" = 1375a8a3-bdd9-41d1-a211-6d51a57d784e AND U0."id" = "discount_promotionrule_channels"."promotionrule_id") LIMIT 1)`

Reply:
Yeah, but firstly you are iterating through all rules and getting their ids